public/app/features/serviceaccounts/ServiceAccountPage.tsx

```diff
@@ -1,31 +1,56 @@
 import React, { PureComponent } from 'react';
 import { connect, ConnectedProps } from 'react-redux';
-import { NavModel } from '@grafana/data';
+import { getTimeZone, NavModel } from '@grafana/data';
 import { getNavModel } from 'app/core/selectors/navModel';
 import Page from 'app/core/components/Page/Page';
 import { ServiceAccountProfile } from './ServiceAccountProfile';
-import { StoreState, ServiceAccountDTO } from 'app/types';
+import { StoreState, ServiceAccountDTO, ApiKey } from 'app/types';
 import { GrafanaRouteComponentProps } from 'app/core/navigation/types';
-import { loadServiceAccount } from './state/actions';
+import { loadServiceAccount, loadServiceAccountTokens, deleteServiceAccountToken } from './state/actions';
+import { ServiceAccountTokensTable } from './ServiceAccountTokensTable';
 
 interface OwnProps extends GrafanaRouteComponentProps<{ id: string }> {
   navModel: NavModel;
   serviceAccount?: ServiceAccountDTO;
+  tokens: ApiKey[];
+}
+
+interface State {
   isLoading: boolean;
 }
 
-export class ServiceAccountPage extends PureComponent<Props> {
+export class ServiceAccountPage extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      isLoading: false,
+    };
+  }
+
   async componentDidMount() {
+    await this.fetchServiceAccount();
+  }
+
+  async fetchServiceAccount() {
     const { match } = this.props;
-    this.props.loadServiceAccount(parseInt(match.params.id, 10));
+    this.setState({ isLoading: true });
```

Comment:
I am not sure about this change, here I have worked with the actions, reducers to introduce the state of the pages that I have worked with.

Sort of like, all the state management of the pages should be in : `/state/*.ts`

Reply:
I got it from the teams page, it might not be the best source :shrug:

Reply:
It's better to use isLoading state from redux, you can set it in reducers when data is loaded.